### PR TITLE
add Uniform Type Identifier to osxpkg

### LINKED ISSUE #5103: Define Uniform Type Identifier for macOS install package.
The macOS install package should adopt a proper [Uniform Type Identifier (UTI)](https://developer.apple.com/library/content/documentation/FileManagement/Conceptual/understanding_utis/understand_utis_conc/understand_utis_conc.html#//apple_ref/doc/uid/TP40001319-CH202-CHDHIJDE). Current this is just `powershell`. `fpm` offers a `--osxpkg-identifier-prefix` which would add a reverse domain such as `com.microsoft`, final UTI would be `com.microsoft.powershell`. Perhaps even the suffix should be different, e.g powershellcore.

/cc @TravisEz13

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Use standard reverse dns format for UTI prefix: com.microsoft

Fix https://github.com/PowerShell/PowerShell/issues/5103

### CODE REVIEW

**Comment by TravisEz13** [comment]
@thezim Please review

**Review by TravisEz13 — APPROVED** [review]

**Review by thezim — APPROVED** [review]
LGTM.

File: tools/packaging/packaging.psm1 [hunk 1]
```diff
@@ -650,6 +650,9 @@ function New-UnixPackage {
             $Arguments += @("--rpm-dist", "rhel.7")
             $Arguments += @("--rpm-os", "linux")
         }
+        if ($Environment.IsMacOS) {
+            $Arguments += @("--osxpkg-identifier-prefix", "com.microsoft")
```
[the remark this answers is not in the document]
Comment (thezim) [reply tools/packaging/packaging.psm1]: Never mind, I forget #5291 has not gone in yet.

**Comment by TravisEz13** [comment]
@thezim Can you update your PR?

**Pull request merged**